packages/analytics-types/src/plugin.ts

```diff
@@ -3,30 +3,29 @@ import { Config } from './config';
 import { Result } from './result';
 import { CoreClient } from './client/core-client';
 
-export enum PluginType {
-  BEFORE = 'before',
-  ENRICHMENT = 'enrichment',
-  DESTINATION = 'destination',
+type BeforePluginType = 'before';
```

Comment:
nit: I would prefix with `PluginType` for easier code completion for end users. e.g. `PluginTypeBefore`, `PluginTypeEnrichment` so you can do `Plugin` + ctrl + space and see all the options.

Reply:
Fixed: https://github.com/amplitude/Amplitude-TypeScript/pull/407/commits/edfd8dc7bf733e2a7ef10e1e3b8fa2d3ed976551